src/movies.js

```diff
@@ -1,22 +1,114 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data");
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+
+  const neW = movies.map((eachMovies)=>{
+    return eachMovies.director;
+  })
+
+  return neW
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+
+    const newArry = movies.filter((eachMovies)=>{
+        return eachMovies.genre.toLowerCase() === 'drama';
+    })
+
+    return eachMovies.genre;
+}
+
+
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+
+  const scoreArr = movies.reduce(function(accu, current){
+   
+    return accu + current.score;
+  },0)
+
+ 
+
+  return Number((scoreArr/movies.length).toFixed(2));
+}
```

Comment:
- Aquí te falta cubrir los casos de los tests en los que si el array que se pasa está vacío, la función devuelva 0, y que si uno de los elementos del array no tiene `score`, aún así devuelva un resultado correcto. 
- Esto último se conseguiría teniendo en cuenta el caso de que exista o no la propiedad `score` en cada elemento sobre el que se itera dentro del `sort()`:

```suggestion
  const scoreArr = movies.reduce((accu, current) => {
    if (current.score) {
      return accu + current.score;
    } else {
      return accu
    }
  }, 0)

```